packages/react-dom/src/__tests__/ReactServerRendering-test.js

```diff
@@ -372,6 +372,22 @@ describe('ReactDOMServer', () => {
       expect(element.firstChild.focus).not.toHaveBeenCalled();
     });
 
+    it('should not focus on either server or client with autofocus={false} even if the markup is mismatch', () => {
```

Comment:
Did you verify this test fails without your fix?

Reply:
Yes.